src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -81,7 +85,10 @@ abstract class AbstractCheckTask(
     private val eventBatchRouter: MessageRouter<EventBatch>
 ) : AbstractSessionObserver<MessageContainer>() {
 
-    protected open class Refs(val rootEvent: Event)
+    protected open class Refs(
+        val rootEvent: Event,
+        val onTaskFinished: ((EventStatus) -> Unit)?
```

Comment:
Why don't we add it to `RuleConfiguration`? I don't see any benefits from setting that parameter in every rule. Also, I think it should not be nullable. We can use an empty lambda by default

Reply:
since we have a container for references, it is appropriate to use it for all references that are not required after the rule is completed

Reply:
Okay, I agree with you. Let's keep it this way